### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by the-redback**

### CODE REVIEW

File: pkg/controller/create.go [hunk 1]
```diff
@@ -86,48 +84,44 @@ func (c *Controller) findStatefulSet(memcached *tapi.Memcached) (bool, error) {
...
-			Labels:      memcached.StatefulSetLabels(),
-			Annotations: memcached.StatefulSetAnnotations(),
+			Labels:      memcached.DeploymentLabels(),
+			Annotations: memcached.DeploymentAnnotations(),
 		},
-		Spec: apps.StatefulSetSpec{
-			Replicas:    types.Int32P(1),
-			ServiceName: c.opt.GoverningService,
+		Spec: apps.DeploymentSpec{
+			Replicas: types.Int32P(memcached.Spec.Replicas),
```
Comment (tamalsaha) [inline pkg/controller/create.go]: @the-redback , missing spec properties:
https://github.com/k8sdb/apimachinery/blob/dd8a840c5e36e1dcc74933f9bc5e6ae8c839b5bc/apis/kubedb/v1alpha1/memcached_types.go#L44

File: pkg/controller/lib.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	}
+
+	if !checkSuccess {
+		return errors.New("Fail to delete Deployments Pods")
+	}
+	// Delete Deployments
+	return c.Client.AppsV1beta1().Deployments(deployment.Namespace).Delete(deployment.Name, nil)
+}
+
+func tryPatchDeployment(c kubernetes.Interface, meta metav1.ObjectMeta, transform func(*apps.Deployment) *apps.Deployment) (result *apps.Deployment, err error) {
```
Comment (tamalsaha) [inline pkg/controller/lib.go]: Use the kutil library

**Comment by tamalsaha** [comment]
@the-redback , hold.Hold() is removed. Is that intentional?

**Comment by the-redback** [comment]
Are we talking about this? 
https://github.com/k8sdb/memcached/pull/2/files#diff-3c862eb54a8e0e161b534e0c67e5379eL101

Yes. It was being problematic to run tests. Other repos don't have it. Furthermore, We call this Run() function from another one named  RunAndHold() which itself containes a hold function. 
https://github.com/k8sdb/memcached/blob/b0437720939e31245b09dad5a6adc95a96e0dea8/pkg/controller/controller.go#L94

@tamalsaha

**Pull request merged**